src/binary/int.rs

```diff
@@ -61,8 +61,8 @@ impl DecodedInt {
     /// Encodes a negative zero as an `Int` and writes it to the provided `sink`.
     /// Returns the number of bytes written.
     ///
-    /// This method is similar to [Self::write_i64]. However, because an i64 cannot represent a negative
-    /// zero, a separate method is required.
+    /// This method is similar to [Self::write]. However, because Rust's native integer types cannot
+    /// represent a negative zero, a separate method is required.
     pub fn write_negative_zero<W: Write>(sink: &mut W) -> IonResult<usize> {
```

Comment:
This (confusingly named) type is the [binary 1.0 int encoding primitive](https://amazon-ion.github.io/ion-docs/docs/binary.html#uint-and-int-fields) rather than [the user-level `int` data type](https://amazon-ion.github.io/ion-docs/docs/binary.html#0x2-and-0x3-int).

The encoding primitive is used to [encode a decimal's coefficient](https://amazon-ion.github.io/ion-docs/docs/binary.html#0x5-decimal):

![image](https://github.com/amazon-ion/ion-rust/assets/611616/bd1ec399-298a-4a49-98ab-668e906996d9)

and the coefficient can be `-0`.